### PR TITLE
[fuzz] fix-up small fuzz bugs in JSON parsing 

### PR DESCRIPTION
**Pull request by asraa**
* fixes red herring header_map_impl_fuzz bug due to use-after-frees
* fixes crash on json parsing a large recursive protobuf struct in access log formatter
* speeds up access log formatter fuzz test minorly by moving headermap and stream_info creation out of the loop

Testing: added test-cases
Fixes:
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=13607
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=19557

Signed-off-by: Asra Ali <asraa@google.com>

### CODE REVIEW

File: source/common/protobuf/utility.cc [hunk 1]
```diff
@@ -426,7 +426,10 @@ std::string MessageUtil::getJsonStringFromMessage(const Protobuf::Message& messa
   std::string json;
   const auto status = Protobuf::util::MessageToJsonString(message, &json, json_options);
   // This should always succeed unless something crash-worthy such as out-of-memory.
-  RELEASE_ASSERT(status.ok(), "");
+  if (!status.ok()) {
```
Comment (mattklein123) [inline source/common/protobuf/utility.cc]: It seems like the comment above needs to be adjusted? Why is this failing? Shouldn't we always be able to covert a message to a JSON string?

Comment (asraa) [reply source/common/protobuf/utility.cc]: Yep, in this case this was an OOM. The message was a protobuf struct that was recursive. (The mutator that produces these test-cases sets a recursive protobuf limit of 100, but this was still enough to OOM). Seems like the same happens in `jsonConvertInternal`

Comment (mattklein123) [reply source/common/protobuf/utility.cc]: Hmm, shouldn't we actually crash then? Or it's not an OOM persay but some kind of max recursion on the conversion? The main thing I'm worried about here is I'm not sure it's safe to throw an exception in all cases where this code is used?

Comment (asraa) [reply source/common/protobuf/utility.cc]: Hmm yeah. It looks like admin endpoint uses this. In the access log case, the string returned from this method is truncated (but only after the failure).

It also seems like the filter metadata is a trusted source, and no one will be overloading it with recursive protobuf structs, so I moved the truncation to stop the failure in to the utility functions for the fuzzer. Harvey also opened this issue: https://github.com/envoyproxy/protoc-gen-validate/issues/95

**Pull request merged**